components/script/dom/blob.rs

```diff
@@ -183,61 +199,46 @@ impl Blob {
                 let _ = filemanager.send(FileManagerThreadMsg::ActivateBlobURL(f.id.clone(), tx, origin.clone()));
 
                 match rx.recv().unwrap() {
-                    Ok(_) => f.id.clone(),
-                    Err(_) => SelectedFileId(Uuid::new_v4().simple().to_string()) // Return a dummy id on error
+                    Ok(_) => return f.id.clone(),
+                    Err(_) => return SelectedFileId(Uuid::new_v4().simple().to_string()) // Return a dummy id on error
                 }
             }
-            BlobImpl::Memory(ref slice) => {
-                self.promote(slice, true)
-            }
-            BlobImpl::Sliced(ref parent, ref rel_pos) => {
-                match *parent.blob_impl.borrow() {
-                    BlobImpl::Sliced(_, _) => {
-                        debug!("Sliced can't have a sliced parent");
-                        // Return dummy id
-                        SelectedFileId(Uuid::new_v4().simple().to_string())
-                    }
-                    BlobImpl::File(ref f) =>
-                        self.create_sliced_url_id(&f.id, rel_pos),
-                    BlobImpl::Memory(ref bytes) => {
-                        let parent_id = parent.promote(bytes, false);
-                        *self.blob_impl.borrow_mut() = BlobImpl::Sliced(parent.clone(), rel_pos.clone());
-                        self.create_sliced_url_id(&parent_id, rel_pos)
-                    }
-                }
-            }
-        }
-    }
+            BlobImpl::Memory(ref bytes) => bytes.clone(),
```

Comment:
we shouldn't clone here, we should move the bytes out. You can use mem::swap and an empty temporary vec to avoid the borrow errors.